beacon-chain/core/light-client/lightclient.go

```diff
@@ -611,64 +454,48 @@ func getExecutionPayloadHeaderDeneb(block interfaces.ReadOnlySignedBeaconBlock)
-		Timestamp:        payloadInterface.Timestamp(),
-		ExtraData:        payloadInterface.ExtraData(),
-		BaseFeePerGas:    payloadInterface.BaseFeePerGas(),
-		BlockHash:        payloadInterface.BlockHash(),
-		TransactionsRoot: transactionsRoot,
-		WithdrawalsRoot:  withdrawalsRoot,
-	}
-
-	return execution, nil
+	return withdrawalsRoot, nil
 }
 
-func ComputeTransactionsRoot(payload interfaces.ExecutionData) ([]byte, error) {
-	transactionsRoot, err := payload.TransactionsRoot()
-	if errors.Is(err, consensus_types.ErrUnsupportedField) {
-		transactions, err := payload.Transactions()
+func BlockToLightClientHeader(block interfaces.ReadOnlySignedBeaconBlock) (*ethpbv2.LightClientHeaderContainer, error) {
+	switch block.Version() {
+	case version.Altair, version.Bellatrix:
+		altairHeader, err := BlockToLightClientHeaderAltair(block)
 		if err != nil {
-			return nil, errors.Wrap(err, "could not get transactions")
+			return nil, errors.Wrap(err, "could not get altair header")
 		}
-		transactionsRootArray, err := ssz.TransactionsRoot(transactions)
-		if err != nil {
-			return nil, errors.Wrap(err, "could not get transactions root")
-		}
-		transactionsRoot = transactionsRootArray[:]
-	} else if err != nil {
-		return nil, errors.Wrap(err, "could not get transactions root")
-	}
-	return transactionsRoot, nil
-}
-
-func ComputeWithdrawalsRoot(payload interfaces.ExecutionData) ([]byte, error) {
-	withdrawalsRoot, err := payload.WithdrawalsRoot()
-	if errors.Is(err, consensus_types.ErrUnsupportedField) {
-		withdrawals, err := payload.Withdrawals()
+		return &ethpbv2.LightClientHeaderContainer{
+			Header: &ethpbv2.LightClientHeaderContainer_HeaderAltair{
+				HeaderAltair: altairHeader,
+			},
+		}, nil
+	case version.Capella:
+		capellaHeader, err := BlockToLightClientHeaderCapella(context.Background(), block)
 		if err != nil {
-			return nil, errors.Wrap(err, "could not get withdrawals")
+			return nil, errors.Wrap(err, "could not get capella header")
 		}
-		withdrawalsRootArray, err := ssz.WithdrawalSliceRoot(withdrawals, fieldparams.MaxWithdrawalsPerPayload)
+		return &ethpbv2.LightClientHeaderContainer{
+			Header: &ethpbv2.LightClientHeaderContainer_HeaderCapella{
+				HeaderCapella: capellaHeader,
+			},
+		}, nil
+	case version.Deneb:
+		denebHeader, err := BlockToLightClientHeaderDeneb(context.Background(), block)
 		if err != nil {
-			return nil, errors.Wrap(err, "could not get withdrawals root")
+			return nil, errors.Wrap(err, "could not get deneb header")
 		}
-		withdrawalsRoot = withdrawalsRootArray[:]
-	} else if err != nil {
-		return nil, errors.Wrap(err, "could not get withdrawals root")
+		return &ethpbv2.LightClientHeaderContainer{
+			Header: &ethpbv2.LightClientHeaderContainer_HeaderDeneb{
+				HeaderDeneb: denebHeader,
+			},
+		}, nil
+	default:
+		return nil, fmt.Errorf("unsupported block version %s", version.String(block.Version()))
 	}
-	return withdrawalsRoot, nil
 }
 
+// TODO: make below functions private
```

Comment:
TODO

Reply:
I probably should do this in a separate PR since I have to change all the usecases and tests.